powerdns/views.py

```diff
@@ -61,17 +70,149 @@ class DomainViewSet(OwnerViewSet):
+            if (
+                self.request._request.method == 'DELETE' and
+                not can_delete(self.request.user, obj)
+            ):
+                raise exceptions.PermissionDenied(detail='No permission')
+        return obj
+
+    def get_serializer_class(self):
+        if self.request.version == 'v1':
+            serializer_class = RecordSerializer
+        if self.request.version == 'v2':
+            serializer_class = RecordSerializerV2
+        return serializer_class
+
+    def create(self, request, *args, **kwargs):
+        if self.request.version == 'v1':
+            return super().create(request, *args, **kwargs)
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        record_request = RecordRequest(
+            domain_id=serializer.validated_data['domain'],
+            owner_id=request.user.id,
+        )
+        record_request.copy_records_data(serializer.validated_data.items())
+        record_request.save()
+
+        auto_accept = (
+            self.request.user.is_superuser or
+            serializer.validated_data['domain'].unrestricted or
+            self.request.user == serializer.validated_data['domain'].owner or
+            request.user.id in serializer.validated_data['domain'].authorisations.values_list('authorised', flat=True) # noqa
+        )
+        if auto_accept:
+            serializer.save()
+            record_request.record = serializer.instance
+            record_request.accept()
+            data = serializer.data
+            code = status.HTTP_201_CREATED
+            headers = {}
+        else:
+            data = {}
+            code = status.HTTP_202_ACCEPTED
+            headers = {
+                'Location': reverse(
+                    'recordrequest-detail', kwargs={'pk': record_request.id}
+                )
+            }
+        return Response(data, status=code, headers=headers)
+
+    def update(self, request, *args, **kwargs):
+        if self.request.version == 'v1':
+            return super().update(request, *args, **kwargs)
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+
+        record_request = RecordRequest(
+            domain_id=serializer.instance.domain_id,
+            owner_id=request.user.id,
+            record_id=serializer.instance.id,
+        )
+        # in validated_data lands fields required by model, even if they
+        # weren't changed, so filter it by matching validated vs initial
+        data_to_copy = [
+            (field_name, serializer.validated_data[field_name])
+            for field_name in (
+                serializer.validated_data.keys() &
+                serializer.initial_data.keys()
+            )
+        ]
+        record_request.copy_records_data(data_to_copy)
```

Comment:
This `copy_records_data` copy data with a bit logic, so I don't see use of this here.